internal/provider/kubernetes/controller.go

```diff
@@ -277,6 +277,14 @@ func (r *gatewayAPIReconciler) Reconcile(ctx context.Context, _ reconcile.Reques
 		return reconcile.Result{}, err
 	}
 
+	gcStatusToDelete := r.loadGatewayClassStatusToDelete()
```

Comment:
can you add a comment here on what this algo does
1. gets all gatewayClass statuses and put it in a set
2. deletes the reconciled gateway classes names from the set
3. deletes the remaining from watchable (representing non existent gateway classes)